framework/include/kernels/ADKernelScalarBase.h

```diff
@@ -0,0 +1,112 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#pragma once
+
+#include "ADKernel.h"
+
+/**
+ * This ADKernel adds standardized methods for assembling to a primary
+ * scalar variable associated with the primary variable of the ADKernel
+ * object. Essentially, the entire row of the residual and Jacobian
+ * associated with this scalar variable will also be assembled here
+ * using the loops over volumetric elements.
+ * This variable is "scalar_variable" in the input file and "kappa"
+ * within the source code.
+ */
+class ADKernelScalarBase : public ADKernel
+{
+public:
+  static InputParameters validParams();
+
+  ADKernelScalarBase(const InputParameters & parameters);
+
+  virtual const MooseVariable & variable() const override { return _var; }
```

Comment:
This should already be in the base class I believe

Reply:
Fixed